### PR TITLE
MMT-1975: add functionality for rejecting proposal workflow

### PR DESCRIPTION
**Pull request by charleshuang80**
* add rejection state, buttons, actions
* add UI elements for rejection including modals and timeline
* add tests
* update timeline
* refactor acl checks
* add migration for proposal default values

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -131,6 +131,8 @@
         get 'submit'
         get 'rescind'
         get 'approve'
+        # TODO: method type? 'reject'
```
Comment (charleshuang80) [inline config/routes.rb]: remove comment

File: spec/features/proposal/access_permissions/permissions_for_non_nasa_draft_approver_spec.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
-describe 'Non-NASA Draft Approver Permissions for Draft MMT', reset_provider: true do
+describe 'Non-NASA Draft Approver Permissions for Draft MMT', reset_provider: true, js: true do
   before do
     set_as_proposal_mode_mmt(with_draft_approver_acl: true)
   end
+  # TODO: any more tests needed here?
```
Comment (charleshuang80) [inline spec/features/proposal/access_permissions/permissions_for_non_nasa_draft_approver_spec.rb]: remove

File: spec/features/proposal/access_permissions/permissions_for_non_nasa_draft_approver_spec.rb [hunk 3]
```diff
@@ -35,6 +36,7 @@
 
       context 'when creating a new collection draft proposals page' do
         before do
+          save_screenshot
```
Comment (charleshuang80) [inline spec/features/proposal/access_permissions/permissions_for_non_nasa_draft_approver_spec.rb]: remove

File: spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    context 'when the rejecting the proposal' do
+      before do
+        click_on 'Reject Proposal Submission'
+      end
+
+      it 'displays the modal to provide rejection feedback' do
+        within '#reject-submission-modal' do
+          expect(page).to have_content('Please provide the reasons for rejecting this proposal submission.')
+          # field for reasons
```
Comment (charleshuang80) [inline spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb]: remove comments

File: app/views/proposal/collection_draft_proposals/show.html.erb [hunk 5]
```diff
@@ -22,15 +22,18 @@
...
+          <%= link_to state_action_button_text('Approve'), "#approve-submission-modal", class: 'eui-btn--blue display-modal' %>
+          <%= link_to state_action_button_text('Reject'), "#reject-submission-modal", class: 'eui-btn--red display-modal' %>
           <%= render partial: 'proposal/collection_draft_proposals/proposal_partials/approve_modal' %>
-          <%= render partial: 'proposal/collection_draft_proposals/proposal_partials/rescind_modal' %>
-        <% else %>
-          <%= link_to state_action_button_text('Cancel'), "#rescind-submission-modal", class: 'eui-btn--blue display-modal' %>
-          <%= render partial: 'proposal/collection_draft_proposals/proposal_partials/rescind_modal' %>
+          <%= render partial: 'proposal/collection_draft_proposals/proposal_partials/reject_modal' %>
         <% end %>
+        <%= link_to state_action_button_text('Cancel'), "#rescind-submission-modal", class: 'eui-btn--blue display-modal' %>
```
Comment (ryanmiller-1) [inline app/views/proposal/collection_draft_proposals/show.html.erb]: This is what I was asking about; should this be a button as an approver?

File: spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        it 'does not navigate away when cancelling' do
+          within '#proposal-status-display' do
+            expect(page).to have_content('Draft Proposal Submission')
+            expect(page).to have_content('Submitted')
+          end
+        end
+      end
+    end
+
+    context 'when the approving fails' do
```
Comment (ryanmiller-1) [inline spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb]: The context and comment should probably be changed to make sense for this file.

File: app/views/proposal/collection_draft_proposals/show.html.erb (hunk 5 above)
Comment (charleshuang80) [reply app/views/proposal/collection_draft_proposals/show.html.erb]: I checked with the PO, she likes it this way for now.

As an aside, she is thinking through if all of the action buttons for the User workflow should be present all the time, with some being disabled depending on the state. You may want to weigh in if you have an opinion about it.

File: spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb (hunk 6 above)
Comment (charleshuang80) [reply spec/features/proposal/collection_draft_proposals/collection_draft_proposal_reject_spec.rb]: addressed

**Pull request merged**